L1Trigger/Phase2L1ParticleFlow/plugins/L1HPSPFTauProducer.cxx

```diff
@@ -0,0 +1,204 @@
+#include <vector>
+#include <numeric>
+
+////////////////////
+// FRAMEWORK HEADERS
+#include "FWCore/Framework/interface/global/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+
+#include "DataFormats/L1TParticleFlow/interface/PFCandidate.h"
+#include "DataFormats/L1TParticleFlow/interface/PFTau.h"
+#include "DataFormats/JetReco/interface/CaloJet.h"
+// bitwise emulation headers
+#include "L1Trigger/Phase2L1ParticleFlow/interface/taus/L1HPSPFTauEmulator.h"
+
+
+class L1HPSPFTauProducer : public edm::global::EDProducer<> {
+    
+public:
+  explicit L1HPSPFTauProducer(const edm::ParameterSet&);
+  ~L1HPSPFTauProducer() override {};
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+  
+
+private:
+  
+  
+  void produce(edm::StreamID, edm::Event& iEvent, const edm::EventSetup& iSetup) const override;
+  
+
+  //various needed vars
+  int _nTaus;
```

Comment:
Use trailing underscore instead (`_` as first character is not allowed, 2.14 in https://cms-sw.github.io/cms_coding_rules.html#2--naming-rules-1)